### ISSUE TITLE
Installation fails due to conflicting botocore version

### ISSUE THREAD

**Issue by NeolithEra** [nficano/python-lambda#500]
Hi, users are unable to run Dmn-python-lambda due to dependency conflict with _**botocore**_ package. As shown in the following full dependency graph of Dmn-python-lambda, Dmn-python-lambda requires _**botocore>=1.5.62,<2.0.0**_，while boto3==1.4.4 requires _**botocore>=1.5.0,<1.6.0**_.

According to pip’s “first found wins” installation strategy, _**botocore 1.16.16**_ is the actually installed version. However, _**botocore 1.16.16**_ does not satisfy _**botocore>=1.5.0,<1.6.0**_.

### Dependency tree-----------
```
dmn-python-lambda - 3.3.1
| +- boto3(install version:1.4.4 version range:==1.4.4)
| | +- botocore(install version:1.5.95 version range:>=1.5.0,<1.6.0)
| | | +- docutils(install version:0.14 version range:>=0.10)
| | | +- jmespath(install version:0.9.5 version range:<1.0.0,>=0.7.1)
| | | +- python-dateutil(install version:2.7.5 version range:>=2.1,<3.0.0)
| | | | +- six(install version:1.11.0 version range:*)
| | | | +- six(install version:1.11.0 version range:>=1.5)
| | +- jmespath(install version:0.9.5 version range:<1.0.0,>=0.7.1)
| | +- s3transfer(install version:0.1.13 version range:>=0.1.10,<0.2.0)
| | | +- botocore(install version:1.16.19 version range:>=1.3.0,<2.0.0)
| +- botocore(install version:1.16.19 version range:>=1.5.62,<2.0.0)
| +- click(install version:6.6 version range:==6.6)
| +- docutils(install version:0.14 version range:<0.15,>=0.12)
| +- futures(install version:3.3.0 version range:*)
| +- jmespath(install version:0.9.5 version range:>=0.9.0,<0.10.0)
| +- pyaml(install version:15.8.2 version range:==15.8.2)
| | +- pyyaml(install version:3.13 version range:*)
| +- python-dateutil(install version:2.7.5 version range:<2.8.0,>=2.5.3)
| | +- six(install version:1.11.0 version range:*)
| | +- six(install version:1.11.0 version range:>=1.5)
| +- pyyaml(install version:3.13 version range:>=3.11,<3.14)
| +- six(install version:1.11.0 version range:<1.12.0,>=1.10.0)
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution

1.	Loosen the version range of **_boto3_** to be **_>=1.4.4_**.
2.	Remove your direct dependency **_botocore_**, and use the **_botocore_** transitively introduced by **_boto3_**.
3.	Loosen the version range of **_botocore_** to be **_>=1.5.62,<1.6.0_**.
@nficano Which solution do you prefer, 1 ,2 or 3?
Please let me know your choice. May I pull a request to solve this issue?

**Comment by zacharyabresch** [comment]
I'm also having this issue.

**Comment by zacharyabresch** [comment]
Is anyone working on this? I believe I can't use Secrets Manager due to this and can't find a way to install the correct versions myself. Would rather not fork.

**Comment by zacharyabresch** [comment]
Or has anyone found a workaround?

**Comment by zacharyabresch** [comment]
Well, I attempted to push a branch for a pull request but I don't have permission (of course). This is really frustrating. I understand this is open-source so I'm not bitching. It would just be nice to get this fixed with such a simple change. There are also a _bunch_ of PRs from Dependabot awaiting review so not sure how active the maintainer will be.

**Comment by zacharyabresch** [comment]
Okay, well, 5 days with no response. That's worrisome to say the least. Guess I'll have to start looking for a new tool

**Comment by NeolithEra** [comment]
@zacharyabresch Sorry for my late reply. Generally, it is a automated tool (implemented as a bot on GitHub https://github.com/apps/watchman-pypi ). But there is administrator looking at your comments. : p 
Welcome try this dependency conflict diagnosis tool for Python projects.

**Comment by cole-seph** [comment]
Plus one on this issue

**Comment by rishabh-bhargava** [comment]
Found some time to fix this (#534 ) - followed bullet points 1 and 2 from the solution mentioned earlier (thanks @NeolithEra). Please use the latest version:  `python-lambda==11.8.0`, which should allow for more up-to-date `boto3` and `botocore` versions. 

If you run into any issues, please let me know - happy to push additional fixes as needed. 

> ### Solution
> 1. Loosen the version range of **_boto3_** to be **_>=1.4.4_**.
> 2. Remove your direct dependency **_botocore_**, and use the **_botocore_** transitively introduced by **_boto3_**.
> 3. Loosen the version range of **_botocore_** to be **_>=1.5.62,<1.6.0_**.
>    @nficano Which solution do you prefer, 1 ,2 or 3?
>    Please let me know your choice. May I pull a request to solve this issue?
